src/dawn/engine_dawn.cc

```diff
@@ -854,7 +985,7 @@ Result EngineDawn::DoDrawRect(const DrawRectCommand* command) {
   static const uint32_t indexData[3 * 2] = {
       0, 1, 2, 0, 2, 3,
   };
-  render_pipeline->index_buffer = CreateBufferFromData(
+  auto index_buffer = CreateBufferFromData(
```

Comment:
When I read this I was concerned about leaking the newly created index buffer (and later the vertex buffer).
But this ok because the ::dawn::Buffer object is destroyed at the end of this method, and the draw command is submitted before then.
So this is fine, it just took a moment for me to realize it.